### PR TITLE
a large number of clippy suggestions

### PR DESCRIPTION
**Pull request by Eh2406**
Just doing small things that clippy told me to.

### CODE REVIEW

File: src/cargo/core/compiler/build_plan.rs [hunk 1]
```diff
@@ -121,11 +121,11 @@ impl BuildPlan {
 
     pub fn update(
         &mut self,
-        invocation_name: String,
-        cmd: ProcessBuilder,
-        outputs: Arc<Vec<OutputFile>>,
```
Comment (oli-obk) [inline src/cargo/core/compiler/build_plan.rs]: This could be just `&[OutputFile]`

File: src/cargo/lib.rs [hunk 2]
```diff
@@ -3,6 +3,14 @@
 // But if someone runs it they should know that
 // @alexcrichton disagree with clippy on some style things
 #![cfg_attr(feature = "cargo-clippy", allow(explicit_iter_loop))]
+#![cfg_attr(feature = "cargo-clippy", allow(explicit_into_iter_loop))]
+// alsow we use closures as a alternative to try catch blocks
```
Comment (oli-obk) [inline src/cargo/lib.rs]: alsow

Comment (oli-obk) [reply src/cargo/lib.rs]: `as a` -> `as an`

File: src/cargo/ops/resolve.rs [hunk 3]
```diff
@@ -160,12 +160,12 @@ pub fn resolve_with_previous<'a, 'cfg>(
         );
     }
 
-    let ref keep = |p: &&'a PackageId| {
+    let keep = &(|p: &&'a PackageId| {
```
Comment (oli-obk) [inline src/cargo/ops/resolve.rs]: this seems unfortunate, can you open an issue about it in the clippy repo?

Comment (oli-obk) [reply src/cargo/ops/resolve.rs]: I think `&(|| { ... })` is not very good. the `ref` seemed a lot prettier. We should not report removing the `ref` in favour of a `&` if the rhs is a closure.

**Comment by Eh2406** [comment]
Ci passing and nits addressed.

**Comment by alexcrichton** [comment]
@bors: r+

Nice!

**Comment by dwijnand** [comment]
Nice! Thanks for taking care of this, Jacob.

> To be clear, I did not do all the things clippy suggested. Just the ones that are small and seemed reasonable.

Just out of interest (#lazyweb) does the sum of the fixes here and the opt-outs added mean clippy runs cleanly now, as in returns no errors or warnings?

**Comment by Eh2406** [comment]
@dwijnand No, we still have ~ 2 pages of nits that clippy suggested. There just more involved things, more like one commit per suggestion.

**Comment by dwijnand** [comment]
OK. Are you planning on taking those on in due time? Just wondering.. I don't want to duplicate work (or worst trod on people's feet 😄).

I'm in no rush, btw, I'm currently working on the E-help-wanted labeled [A-testing-cargo-itself](https://github.com/rust-lang/cargo/labels/A-testing-cargo-itself) issues (https://github.com/rust-lang/cargo/issues/5744 currently).

**Comment by Eh2406** [comment]
It is all yours! I'd love to see PR's that pick them off, or that allow each one in a targeted way with a comment about why.

**Comment by ehuss** [comment]
Does anyone know why `do_not_package_if_src_was_modified` fails on mac?  I'm able to repro ~30% of the time.

**Comment by ehuss** [comment]
@Eh2406 It looks like it has been that way since the test was introduced (it's unrelated to these changes).  I can take a look at it.  Usually it's because macos has a 1s resolution in the filesystem, but in this case I'm a little surprised since a *new* file should invalidate the fingerprint.

**Comment by Eh2406** [comment]
In that case

@bors retry

**Pull request merged**